src/material/button/button.ts

```diff
@@ -48,8 +50,10 @@ import {
   exportAs: 'matButton',
   encapsulation: ViewEncapsulation.None,
   changeDetection: ChangeDetectionStrategy.OnPush,
+  hostDirectives: [MatButtonRipple],
```

Comment:
I was under the impression that we also wanted to avoid creating a new directive instance for button ripples since it has a slight perf overhead.

Reply:
There was no way to avoid fully removing the directive because we expose the ripple to users from our public api. So we still need to provide a `MatRipple` like object to users if they try to interact with it programmatically

The closest we could get to removing it was to create a mock ripple (`MatButtonRipple`) which doesn't actually instantiate the ripple until a dev tries to reference it

Once the `ripple` is removed from the public API, we will be able to completely remove this directive

Reply:
I don't get, though, why this needs to be a host directive instead of something in a service

Reply:
I was able to work around this by just calling the `MatRipple`'s constructor directly and managed to get things working.